code/modules/reagents/oldchem/reagents/reagents_admin.dm

```diff
@@ -51,4 +52,24 @@
 /datum/reagent/adminordrazine/nanites
 	name = "Nanites"
 	id = "nanites"
-	description = "Nanomachines that aid in rapid cellular regeneration."
\ No newline at end of file
+	description = "Nanomachines that aid in rapid cellular regeneration."
+
+
+// For random item spawning. Takes a list of paths, and returns the same list without anything that contains admin only reagents
+
+/proc/adminReagentCheck(var/list/incoming)
+	var/list/outgoing[0]
+	for(var/tocheck in incoming)
+		if(ispath(tocheck))
+			var/check = new tocheck
+			if (istype(check, /atom))
+				var/atom/reagentCheck = check
+				var/datum/reagents/reagents = reagentCheck.reagents
+				var/admin = 0
+				for(var/datum/reagent/reagent in reagents.reagent_list)
```

Comment:
@Fox-McCloud 
Yeah, I understand. One question though, where do you draw the line on sanity checks? For example, do I take out the istype check at line 65? There is no reason I can see to pass datum paths into a proc involved in random item spawning, which is the only thing that would trip it up if I take that out, but with some of the bugs I've fixed and code I've seen while fixing them, I tend to be reluctant to trust that it'll always be used correctly